### PR TITLE
fix: re-enable warning and address 20 warnings

### PR DESCRIPTION
**Pull request by poppoerika**
This pr address the first 20 warnings of 165 warning we have for documentation.
Relevant ticket: https://github.com/momentohq/client-sdk-dotnet/issues/223

### CODE REVIEW

File: src/Momento.Sdk/Auth/ICredentialProvider.cs [hunk 1]
```diff
@@ -2,9 +2,21 @@ namespace Momento.Sdk.Auth;
...
+    /// </summary>
     string AuthToken { get; }
-    string ControlEndpoint {get; }
+    /// <summary>
+    /// Control endpoint extracted from JWT
+    /// </summary>
+    string ControlEndpoint { get; }
+    /// <summary>
+    /// Cache endpoint extracted from JWT
+    /// </summary>
```
Comment (pgautier404) [inline src/Momento.Sdk/Auth/ICredentialProvider.cs:20]: In the future, the endpoints won't be embedded in the JWT, so these interface comments should probably be more general. I'd suggest we just use something like "Momento authentication token", "Control endpoint", and "Cache endpoint" as the summaries for these.

Comment (poppoerika) [reply src/Momento.Sdk/Auth/ICredentialProvider.cs:20]: 👍 
As I was adding comments, I wasn't sure how descriptive we want those properties to be, especially these are hard to describe more than "auth token" and "endpoints" 😄

File: src/Momento.Sdk/Config/Configurations.cs [hunk 2]
```diff
@@ -5,6 +5,9 @@
 
 namespace Momento.Sdk.Config;
 
+/// <summary>
+/// Provide pre-build configurations.
```
Comment (cprice404) [inline src/Momento.Sdk/Config/Configurations.cs:9]: s/build/built

File: src/Momento.Sdk/Auth/EnvMomentoTokenProvider.cs [hunk 3]
```diff
@@ -4,10 +4,22 @@ namespace Momento.Sdk.Auth;
...
+    /// JWT provided by user
+    /// </summary>
     public string AuthToken { get; private set; }
-    public string ControlEndpoint {get; private set; }
+    /// <summary>
+    /// Control endpoint extracted from JWT
+    /// </summary>
+    public string ControlEndpoint { get; private set; }
+    /// <summary>
+    /// Cache endpoint extracted from JWT
```
Comment (cprice404) [inline src/Momento.Sdk/Auth/EnvMomentoTokenProvider.cs:21]: For these two I think you can just use inheritdoc to get the docs for whatever we put on the interface (@malandis @pgautier404  correct me if that sounds wrong.)

File: src/Momento.Sdk/Auth/ICredentialProvider.cs [hunk 4]
```diff
@@ -2,9 +2,21 @@ namespace Momento.Sdk.Auth;
 
 using System.Collections.Generic;
 
+/// <summary>
+/// Interface for credentials
```
Comment (cprice404) [inline src/Momento.Sdk/Auth/ICredentialProvider.cs:6]: Best practice for documentation strings:

If your docstring uses the same words as the definition/name, then it's not really adding any value.  So e.g. you don't need to say `interface` here, that will already be visible to the user.

Maybe say something like:

Specifies the fields that are required for a Momento client to connect to and authenticate with the Momento service.

**Review by pgautier404 — APPROVED** [review]

File: src/Momento.Sdk/Auth/ICredentialProvider.cs [hunk 5]
```diff
@@ -2,9 +2,21 @@ namespace Momento.Sdk.Auth;
...
+/// </summary>
 public interface ICredentialProvider
 {
+    /// <summary>
+    /// JWT provided by user
+    /// </summary>
     string AuthToken { get; }
-    string ControlEndpoint {get; }
+    /// <summary>
+    /// Control endpoint
```
Comment (cprice404) [inline src/Momento.Sdk/Auth/ICredentialProvider.cs:15]: same comment here; this doesn't give any new information beyond what the name would already provide to the user.

Here I think you might want to say something like "the host and port on which the Momento client will connect to the Momento control plane, in the format '<host>:<port>'" or something. 

(I don't know if that's the actual format, you should base64-decode that segment of your JWT and verify that before updating the docs.

File: src/Momento.Sdk/Config/Configuration.cs [hunk 6]
```diff
@@ -8,20 +8,39 @@
 
 namespace Momento.Sdk.Config;
 
+/// <inheritdoc cref="Momento.Sdk.Config.IConfiguration" />
```
Comment (cprice404) [inline src/Momento.Sdk/Config/Configuration.cs:11]: 👍

File: src/Momento.Sdk/Config/Configuration.cs [hunk 7]
```diff
@@ -8,20 +8,39 @@
...
 public class Configuration : IConfiguration
 {
+    /// <inheritdoc cref="Microsoft.Extensions.Logging.ILoggerFactory" />
```
Comment (cprice404) [inline src/Momento.Sdk/Config/Configuration.cs:14]: I like this, if it works the way we want it to.  @malandis is this a reasonable thing to do?  Annotating a Property with an inheritdoc for a Class?

File: src/Momento.Sdk/Config/Configuration.cs [hunk 8]
```diff
@@ -33,26 +52,51 @@ public Configuration(IRetryStrategy retryStrategy, IList<IMiddleware> middleware
         this.TransportStrategy = transportStrategy;
     }
 
+    /// <summary>
+    ///  Configures logging
```
Comment (cprice404) [inline src/Momento.Sdk/Config/Configuration.cs:56]: This is technically a copy constructor so we should probably allude to that.

"Creates a new instance of the Configuration object, updated to use the specified logger factory."

File: src/Momento.Sdk/Config/Configuration.cs [hunk 9]
```diff
@@ -33,26 +52,51 @@ public Configuration(IRetryStrategy retryStrategy, IList<IMiddleware> middleware
...
+    /// </summary>
+    /// <param name="loggerFactory">This is responsible for configuraing logging.</param>
+    /// <returns>Configuration object with custom logging provided</returns>
     public IConfiguration WithLoggerFactory(ILoggerFactory loggerFactory)
     {
         return new Configuration(RetryStrategy, Middlewares, TransportStrategy, loggerFactory);
     }
 
+    /// <summary>
+    ///  Configures retry strategy
```
Comment (cprice404) [inline src/Momento.Sdk/Config/Configuration.cs:66]: Same comment for the rest of these copy constructors.

**Review by cprice404 — COMMENTED** [review]
thanks for doing this!  left a few small nits to help you level up on some best practices for writing docstrings.  Overall looks great though!

**Pull request merged**